proto/v1/idp_service.proto

```diff
@@ -124,6 +131,34 @@ message UndeleteIdentityProviderRequest {
   string name = 1 [(google.api.field_behavior) = REQUIRED];
 }
 
+message TestIdentityProviderRequest {
+  // The identity provider to test connection including uncreated.
+  IdentityProvider identity_provider = 1;
+
+  oneof context {
+    OAuth2IdentityProviderTestRequestContext oauth2_context = 2;
+  }
+}
+
+message OAuth2IdentityProviderTestRequestContext {
+  // Authorize code from website.
+  string code = 1;
+}
+
+message TestIdentityProviderResponse {
+  string message = 1;
```

Comment:
What's this message?

Reply:
The `message` field is using to save the reason of test failure. 

It's possible that we got `access_token', but failed to get `userinfo'. So I don't think  embedding the failure result in RPC status can do that.